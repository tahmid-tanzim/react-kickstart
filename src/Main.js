import React, {Component} from 'react';
import {BrowserRouter as Router, Route, Redirect, Switch} from 'react-router-dom';

import PrivateLayout from './components/PrivateLayout';
import PublicLayout from './components/PublicLayout';

import Home from './components/Home';
import CampaignList from './components/CampaignList';
import Login from './components/Login';
import NotFound from './components/NotFound';

const PrivateRoute = ({component: Component, ...rest}) => {
    return (
        <Route {...rest} render={matchProps => (
            <PrivateLayout>
                <Component {...matchProps} />
            </PrivateLayout>
        )}/>
    );
};

const PublicRoute = ({component: Component, ...rest}) => {
    return (
        <Route {...rest} render={matchProps => (
            <PublicLayout>
                <Component {...matchProps} />
            </PublicLayout>
        )}/>
    );
};

class Main extends Component {
    render() {
        return (
            <Router>
                <Switch>
                    <Route exact path="/">
                        <Redirect to="/login"/>
                    </Route>
                    <PublicRoute path="/login" component={Login}/>
                    <PrivateRoute path="/app/home" component={Home}/>
                    <PrivateRoute path="/app/campaigns" component={CampaignList}/>
                    <Route component={NotFound}/>
                </Switch>
            </Router>
        );
    }
}

export default Main;
