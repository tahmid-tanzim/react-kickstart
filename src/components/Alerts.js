import React from 'react';

const Alerts = props => {
    return (
        <div className={`alert alert-dismissible alert-${props.type}`} role="alert">
            <button type="button" className="close" data-dismiss="alert" aria-label="Close"><span
                aria-hidden="true">&times;</span></button>
            {/*<strong>{ props.type.toUpperCase() }</strong>*/}
            {props.value}
        </div>
    );
};

export default Alerts;