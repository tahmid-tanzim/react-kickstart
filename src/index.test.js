import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Main from './Main';
import Alerts from './components/Alerts';

describe('Main', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Main/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Main />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe('Alerts', () => {

    const props = {
        type: 'success',
        value: 'Campaign added successfully'
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Alerts {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Alerts {...props} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});