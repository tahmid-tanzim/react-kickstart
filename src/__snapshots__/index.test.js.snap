// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts has a valid snapshot 1`] = `
<div
  className="alert alert-dismissible alert-success"
  role="alert"
>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  Campaign added successfully
</div>
`;

exports[`Main has a valid snapshot 1`] = `
<div
  className="container"
>
  <div
    className="col-md-4 col-md-offset-4"
  >
    <div
      className="wrapper-box"
    >
      <form
        className="form-signin"
      >
        <h2
          className="form-signin-heading"
        >
          Please Login
        </h2>
        <input
          autoFocus=""
          className="form-control"
          name="username"
          placeholder="Email Address"
          required=""
          type="text"
        />
        <input
          className="form-control"
          name="password"
          placeholder="Password"
          required=""
          type="password"
        />
        <label
          className="checkbox"
        >
          <input
            disabled={true}
            id="rememberMe"
            name="rememberMe"
            type="checkbox"
            value="remember-me"
          />
           Remember me
        </label>
        <a
          className="btn btn-lg btn-success btn-block"
          href="/app/home"
          onClick={[Function]}
        >
          Login
        </a>
      </form>
    </div>
  </div>
</div>
`;
